### PR TITLE
Allow calling `extra`

### PR DESCRIPTION
**Pull request by operte**
This modification allows the user to call `extra` as the root for all the variables. 

For example, given the following file structure: 
```
mkdocs.yaml
docs/
    1_foo.md
    assets/
        1_foo.yaml
```

`mkdocs.yaml`:
```yaml
plugins:
    - search
    - markdownextradata:
        data: docs/assets
```

`1_foo.yaml`:
```yaml
bar: 42
```

The user can call the variable `bar` as:

`1_foo.md`:
```md
The answer to everything is {{extra['1_foo']['bar']}}
```

Up to now, the user would have to call the variable with `{{1_foo.bar}}`, which raises a `jinja2.exceptions.TemplateSyntaxError`, since the name `1_foo` is not according to the Python variable naming restrictions.

One unintended effect of this modification is that the user is now able to call all entries in `self.mkdocsConfig`. If the user declares a file with a name which is the same as a pre-existing key on `self.mkdocsConfig`, this is not a problem: due to the order in which the dictionaries are merged on line 109, the entries inside `self.mkdocsConfig.get("extra")` overwrite whatever was in `self.mkdocsConfig`.

This solves #34.

### CODE REVIEW

**Comment by operte** [comment]
I updated the documentation to inform the user about this option.

In addition, `apply_template` now catches the `TemplateSyntaxError` that `jinja` raises when the file or foldername is not valid. It informs the users about the `extra` option, refers them to the documentation, and raises the `TemplateSyntaxError` anyway, since the error might have been raised for another reason.

File: README.md [hunk 1]
```diff
@@ -91,16 +91,27 @@ plugins:
...
+For example, if you have a file called `[path/to/datafiles/]sections/captions.yaml` 
+which includes a variable `foo` - where `[path/to/datafiles/]` is the path declared
+in your configuration under `data` - the data inside that file will be available in
+ your templates as `{{sections.captions.foo}}` or `{{sections['captions']['foo']}}`.
+
+Alternatively, you can access all files and variable declared under `data` in template
+using `extra` key. 
+This is particularly useful if your folder or filename do not comply with the Python
+variable naming rules.
+For example, if you have a file `[path/to/datafiles/]1_example/captions.yaml`
```
Comment (rosscdh) [inline README.md:109]: maybe just 

```
path/to/datafiles/1_example/captions.yaml
```

the [] makes difficult to parse

**Review by rosscdh — CHANGES_REQUESTED** [review]
looks good just 1 note, and please add your usecase to the tests

File: README.md (hunk 1 above)
Comment (operte) [reply README.md:109]: I wrote it with `[]` to keep it consistent with the way you had already written the documentation. Are you suggesting removing it on line 109 but keeping it on lines 100 and 101?

Comment (rosscdh) [reply README.md:109]: right you are, strange i dont usually use that notation. naja!

File: test/test_basic.py [hunk 2]
```diff
@@ -23,6 +23,8 @@ def test_basic_working():
         contents = index_file.read_text()
 
         assert 'Where we should see junk.a_key: has a value' in contents, f"docs/_data.junk.yaml does not have the right key/val junk.a_key"
+        assert "Which can also be written as junk['a_key']: has a value" in contents, f"docs/_data.junk.yaml does not have the right key/val junk.a_key"
```
Comment (rosscdh) [inline test/test_basic.py:26]: this is good but i meant more like; add your doc structure with the 1_file into the test structure at 
```
https://github.com/rosscdh/mkdocs-markdownextradata-plugin/tree/master/test/docs
```
so people can actually see a working example of your specific implementation

Comment (operte) [reply test/test_basic.py]: You were faster than me :) I just submitted another change.

Comment (rosscdh) [reply test/test_basic.py]: beautiful mate! :) 💯

**Comment by rosscdh** [comment]
Please add yourself to the README.md as contributors :) and this can go out as the next release

**Review by rosscdh — APPROVED** [review]
once readme is updated with your names ill merge and release

**Comment by operte** [comment]
Ah yes let me add Tim as well, since he helped think this through.

**Comment by rosscdh** [comment]
Thanks for your contrib ❤️

**Comment by operte** [comment]
Hey Ross, can you give us an estimate of when there will be a new release? :)

**Comment by rosscdh** [comment]
https://pypi.org/project/mkdocs-markdownextradata-plugin/

last Saturday evening ;)

**Comment by operte** [comment]
Oh cool, for some reason I don't see the newer releases here on Github. Thanks a lot! :)

**Pull request merged**